Source/panels/charpanel.cpp

```diff
@@ -0,0 +1,325 @@
+		        return StyledText { UiFlags::ColorGold, _("None") };
+	        } else {
+		        return StyledText { UiFlags::ColorSilver, fmt::format("{:d}", MyPlayer->_pNextExper) };
+	        }
+	    } },
+
+	{ N_("Gold"), { 208, 129 }, 87, { 0, -20 }, 0, 0, 1, true, false,
+	    []() { return StyledText { UiFlags::ColorSilver, fmt::format("{:d}", MyPlayer->_pGold) }; } },
+
+	{ N_("Armor class"), { 250, 166 }, 45, { -3, -5 }, 55, 0, 1, false, false,
+	    []() { return StyledText { GetValueColor(MyPlayer->_pIBonusAC), fmt::format("{:d}", MyPlayer->GetArmor()) }; } },
+
+	{ N_("To hit"), { 250, 194 }, 45, { -3, 0 }, 0, 0, 1, false, false,
+	    []() { return StyledText { GetValueColor(MyPlayer->_pIBonusToHit), fmt::format("{:d}%", (MyPlayer->InvBody[INVLOC_HAND_LEFT]._itype == ITYPE_BOW ? MyPlayer->GetRangedToHit() : MyPlayer->GetMeleeToHit())) }; } },
+
+	{ N_("Damage"), { 250, 222 }, 45, { -3, 0 }, 0, 0, 0, false, false,
+	    []() {
+	        std::pair<int, int> dmg = GetDamage();
+	        return StyledText { GetValueColor(MyPlayer->_pIBonusDam), fmt::format("{:d}-{:d}", dmg.first, dmg.second) };
+	    } },
+
+	{ N_("Resist magic"), { 250, 259 }, 45, { -3, -5 }, 46, 1, 1, false, false,
+	    []() { return GetResistInfo(MyPlayer->_pMagResist); } },
+
+	{ N_("Resist fire"), { 250, 287 }, 45, { -3, -5 }, 46, 1, 1, false, false,
+	    []() { return GetResistInfo(MyPlayer->_pFireResist); } },
+
+	{ N_("Resist lightning"), { 250, 315 }, 45, { -3, -5 }, 76, 0, 1, false, false,
+	    []() { return GetResistInfo(MyPlayer->_pLghtResist); } },
+
+};
+
+Art PanelParts[6];
+Art PanelFull;
+
+void DrawPanelFieldLow(const Surface &out, Point pos, int len)
+{
+	DrawArt(out, pos, &PanelParts[0]);
+	pos.x += PanelParts[0].w();
+	DrawArt(out, pos, &PanelParts[1], 0, len);
+	pos.x += len;
+	DrawArt(out, pos, &PanelParts[2]);
+}
+
+void DrawPanelFieldHigh(const Surface &out, Point pos, int len)
+{
+	DrawArt(out, pos, &PanelParts[3]);
+	pos.x += PanelParts[3].w();
+	DrawArt(out, pos, &PanelParts[4], 0, len);
+	pos.x += len;
+	DrawArt(out, pos, &PanelParts[5]);
+}
+
+void DrawShadowString(const Surface &out, Point pos, PanelEntry &entry)
+{
+	if (entry.label == "")
+		return;
+
+	std::string text_tmp = _(entry.label.c_str());
+	char buffer[32];
+	int spacing = entry.labelSpacing;
+	strcpy(buffer, text_tmp.c_str());
+	if (entry.labelLength > 0)
+		WordWrapGameString(buffer, entry.labelLength, GameFontSmall, spacing);
+	std::string text(buffer);
+	int width = GetLineWidth(text, GameFontSmall, spacing);
+	if (entry.centered)
+		width = entry.length;
+	else
+		pos.x -= width;
+
+	Point finalPos = { pos + entry.position + Displacement { 0, 17 } + entry.labelOffset };
+	UiFlags style = UiFlags::AlignRight;
+	if (entry.centered) {
+		style = UiFlags::AlignCenter;
+		finalPos += Displacement { 7, 0 }; // left border
+	}
+	DrawString(out, text, { finalPos + Displacement { -2, 2 }, { width, 0 } }, style | UiFlags::ColorBlack, spacing, 10);

```

Comment:
I can't tell for sure but my impression is that the shadow's (vertical) displacement is 1px greater than the original.

```suggestion
	DrawString(out, text, { finalPos + Displacement { -2, 1 }, { width, 0 } }, style | UiFlags::ColorBlack, spacing, 10);
```

Reply:
no